### ISSUE TITLE
Added localization to PriceV2

### ISSUE THREAD

**Issue by vptcnt** [imsujan276/shopify_flutter#37]
Hello @imsujan276 

I have updated PriceV2 to use the localization of NumberFormat.
It allows displaying correctly the currency symbol before or after the price according to the locale.

/lib/models/src/product/price_v_2/price_v_2.dart
```dart
@freezed
class PriceV2 with _$PriceV2 {

  [...]

  String get formattedPrice => JsonHelper.chooseRightOrderOnCurrencySymbol(
        amount,
        currencyCode,
        priceFormat: priceFormat,
      );

  String formattedPriceWithLocale(String? locale) =>
      JsonHelper.chooseRightOrderOnCurrencySymbol(
        amount,
        currencyCode,
        priceFormat: priceFormat,
        locale: locale,
      );
}
```

I have to add the optional parameter to a json helpe in 
/lib/models/json_helper.dart

```dart
static String chooseRightOrderOnCurrencySymbol(
    dynamic amount,
    String currencyCode, {
    NumberFormat? priceFormat,
    String? locale,  // NEW
  }) {
    return NumberFormat.currency(
      locale: locale,
      name: currencyCode, // NEW
      symbol: '${_simpleCurrencySymbols[currencyCode]}',
    ).format(amountFromJson(amount));
  }
```

and then added two functions at the model Product th 
/lib/models/src/product/product.dart
```dart

@freezed
@JsonSerializable()
class Product with _$Product {

  [...]

  String formattedPriceWithLocale(String? locale) => productVariants.isEmpty
      ? ''
      : productVariants.first.price.formattedPriceWithLocale(locale);

  String compareAtPriceFormattedWithLocale(String? locale) => productVariants.isEmpty
      ? ''
      : (productVariants.first.compareAtPrice == null
          ? ''
          : productVariants.first.compareAtPrice!.formattedPriceWithLocale(locale));

  [...]

}
```

Vince

**Comment by OliverMoller** [comment]
How do i install this on my app?

**Comment by OliverMoller** [comment]
@vptcnt How do i install this on my app?

**Comment by vptcnt** [comment]
@imsujan276 need to add the code to the plugin

**Comment by imsujan276** [comment]
Thank you @vptcnt for this. It helps stores to display the price format according to their locale. 

It will be available in v1.5.6
